app/user.go

```diff
@@ -375,7 +375,10 @@ func (a *App) CreateOAuthUser(service string, userData io.Reader, teamID string,
 		if userByEmail.AuthService == "" {
 			return nil, model.NewAppError("CreateOAuthUser", "api.user.create_oauth_user.already_attached.app_error", map[string]interface{}{"Service": service, "Auth": model.USER_AUTH_SERVICE_EMAIL}, "email="+user.Email, http.StatusBadRequest)
 		}
-		return nil, model.NewAppError("CreateOAuthUser", "api.user.create_oauth_user.already_attached.app_error", map[string]interface{}{"Service": service, "Auth": userByEmail.AuthService}, "email="+user.Email, http.StatusBadRequest)
+		if provider.VerifySameUser(userByEmail, user) {
```

Comment:
Couple of questions:
* What's the benefit of this being in a method vs `if userByEmail.AuthData == user.AuthData`?
* If it's going to be a method can it be renamed such that it's obvious what the truth case is, for example `IsSameUser` or `HasSameAuthDataValue`?
* Is Gitlab the only OAuth provider that needs to implement it?
* Are there any tests that can be updated to reflect this newly found scenario?

Reply:
Yes...I renamed this at least 4 times. I will go with `IsSameUser`, if there are no objections.

Reply:
> Yes...I renamed this at least 4 times. I will go with `IsSameUser`, if there are no objections.

Sounds good to me.